microbench/src/main/java/io/netty/microbench/handler/ssl/AbstractSslEngineBenchmark.java

```diff
@@ -0,0 +1,279 @@
+    public String cipher;
+
+    protected SSLEngine clientEngine;
+    protected SSLEngine serverEngine;
+
+    private ByteBuffer cTOs;
+    private ByteBuffer sTOc;
+    private ByteBuffer serverAppReadBuffer;
+    private ByteBuffer clientAppReadBuffer;
+    private ByteBuffer empty;
+
+    protected final void initEngines() {
+        clientEngine = newClientEngine();
+        serverEngine = newServerEngine();
+    }
+
+    protected final void destroyEngines() {
+        ReferenceCountUtil.release(clientEngine);
+        ReferenceCountUtil.release(serverEngine);
+    }
+
+    protected final void initHandshakeBuffers() {
+        cTOs = allocateBuffer(clientEngine.getSession().getPacketBufferSize());
+        sTOc = allocateBuffer(serverEngine.getSession().getPacketBufferSize());
+
+        serverAppReadBuffer = allocateBuffer(
+                serverEngine.getSession().getApplicationBufferSize());
+        clientAppReadBuffer = allocateBuffer(
+                clientEngine.getSession().getApplicationBufferSize());
+        empty = allocateBuffer(0);
+    }
+
+    protected final void destroyHandshakeBuffers() {
+        freeBuffer(cTOs);
+        freeBuffer(sTOc);
+        freeBuffer(serverAppReadBuffer);
+        freeBuffer(clientAppReadBuffer);
+        freeBuffer(empty);
+    }
+
+    protected final void doHandshake() throws SSLException {
+        clientEngine.beginHandshake();
+        serverEngine.beginHandshake();
+
+        SSLEngineResult clientResult;
+        SSLEngineResult serverResult;
+
+        boolean clientHandshakeFinished = false;
+        boolean serverHandshakeFinished = false;
+
+        do {
+            int cTOsPos = cTOs.position();
+            int sTOcPos = sTOc.position();
+
+            if (!clientHandshakeFinished) {
+                clientResult = clientEngine.wrap(empty, cTOs);
+                runDelegatedTasks(clientResult, clientEngine);
+                assert empty.remaining() == clientResult.bytesConsumed();
+                assert cTOs.position() - cTOsPos == clientResult.bytesProduced();
+
+                if (isHandshakeFinished(clientResult)) {
+                    clientHandshakeFinished = true;
+                }
+            }
+
+            if (!serverHandshakeFinished) {
+                serverResult = serverEngine.wrap(empty, sTOc);
+                runDelegatedTasks(serverResult, serverEngine);
+                assert empty.remaining() == serverResult.bytesConsumed();
+                assert sTOc.position() - sTOcPos == serverResult.bytesProduced();
+
+                if (isHandshakeFinished(serverResult)) {
+                    serverHandshakeFinished = true;
+                }
+            }
+
+            cTOs.flip();
+            sTOc.flip();
+
+            cTOsPos = cTOs.position();
+            sTOcPos = sTOc.position();
+
+            if (!clientHandshakeFinished) {
+                int clientAppReadBufferPos = clientAppReadBuffer.position();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Remove this useless assignment to local variable "clientAppReadBufferPos". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=squid%3AS1854)